stepconfig/stepconfig.go

```diff
@@ -0,0 +1,169 @@
+package stepconfig
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/bitrise-io/go-utils/log"
+)
+
+// ErrInvalidConfig indicates that a configuration is of the wrong type.
+var ErrInvalidConfig = errors.New("conf must be a struct pointer")
+
+// ParseError occurs when a struct field cannot be set.
+type ParseError struct {
+	Field string
+	Value string
+	Err   error
+}
+
+// Error implements builtin errors.Error
+func (e *ParseError) Error() string {
+	segments := []string{e.Field}
+	if e.Value != "" {
+		segments = append(segments, e.Value)
+	}
+	segments = append(segments, e.Err.Error())
+	return strings.Join(segments, ": ")
+}
+
+// Secret variables are not shown in the printed output
+type Secret string
+
+// String implements fmt.Stringer.String.
+// When a Secret is printed, it's masking the underlying string with asterisks.
+func (s Secret) String() string {
+	return strings.Repeat("*", 5)
+}
+
+// Print the name of the struct in blue color followed by a newline,
+// then print all fields and their respective values in separate lines.
```

Comment:
then print all fields and their respective values in separate lines.

->

then print all 'field name: field value' couples separated by new lines.